codegen-core/src/test/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/BuilderGeneratorTest.kt

```diff
@@ -140,4 +143,30 @@ internal class BuilderGeneratorTest {
         }
         project.compileAndTest()
     }
+
+    @Test
+    fun `don't add serde to error types`() {
+        val provider = testSymbolProvider(model)
+        val project = TestWorkspace.testProject(provider)
+        project.moduleFor(errorStruct) {
+            rust("##![allow(deprecated)]")
+            StructureGenerator(model, provider, this, errorStruct, emptyList()).render()
+            implBlock(provider.toSymbol(errorStruct)) {
+                BuilderGenerator.renderConvenienceMethod(this, provider, errorStruct)
+            }
+        }
+        project.withModule(provider.moduleForBuilder(errorStruct)) {
+            BuilderGenerator(model, provider, errorStruct, emptyList()).render(this)
+        }
+        project.compileAndTest()
+
+        // checks if there is a serde derive in the code
+        project.generatedFiles().forEach {
+            if (it.extension == "rs") {
+                val file = project.baseDir.resolve(it).toFile().readText()
+                val check = file.contains("derive(serde::Deserialize)") || file.contains("derive(serde::Serialize)")
```

Comment:
This will catch more issues (note: you'll have to import `assertFalse` from junit):
```suggestion
                assertFalse(file.contains("serde::"))
```
You generally don't want to use the built-in `assert` in unit tests because the JVM disables them by default (so that it won't actually check anything unless you have a system property set to enable it, which we don't).